From 0b71b418293f46c05be78e1afe5efaa31de409f9 Mon Sep 17 00:00:00 2001
From: Lucas Hoffmann <lucc@posteo.de>
Date: Fri, 29 Nov 2019 06:25:09 +0100
Subject: [PATCH] Fix SyntaxWarning for python 3.8.0

Closes #488
---
 multibootusb | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/multibootusb b/multibootusb
index 0d8f2f1..9bd5baf 100644
--- a/multibootusb
+++ b/multibootusb
@@ -196,18 +196,18 @@ def main():
 
     if gui is False:
         check_admin()
-        if uninstall is True and config.usb_disk is not '':
+        if uninstall is True and config.usb_disk != '':
             cli_uninstall_distro()
-        elif uninstall is True and config.usb_disk is '':
+        elif uninstall is True and config.usb_disk == '':
             log('\nYou must provide \'-t\' option to point to your USB disk for uninstalling a distro.\n'
                 'See the usage example below.')
             usage()
-        elif config.image_path is '' and config.usb_disk is '':
+        elif config.image_path == '' and config.usb_disk == '':
             log('\nNo option provided. See the usage below.')
             usage()
-        elif config.cli_syslinux is True and config.usb_disk is not '':
+        elif config.cli_syslinux is True and config.usb_disk != '':
             cli_install_syslinux()
-        elif config.image_path is '' or config.usb_disk is '':
+        elif config.image_path == '' or config.usb_disk == '':
             log('\nOptions \'-i\' and \'-t\' must be supplied together. See the usage below.')
             usage()
         elif config.cli_dd is True:
