diff -up openchange-openchange-2.3-VULCAN/configure.ac.autoconf-2.71 openchange-openchange-2.3-VULCAN/configure.ac
--- openchange-openchange-2.3-VULCAN/configure.ac.autoconf-2.71	2021-03-26 09:10:09.815000000 +0100
+++ openchange-openchange-2.3-VULCAN/configure.ac	2021-03-26 09:10:45.765000000 +0100
@@ -4,7 +4,8 @@
 AC_PREREQ(2.57)
 AC_INIT(openchange, 2.3, [openchange@openchange.org])
 AC_CONFIG_HEADER([config.h])
-AM_INIT_AUTOMAKE
+AC_CONFIG_FILES([Makefile])
+AM_INIT_AUTOMAKE([foreign])
 AC_DEFINE(_GNU_SOURCE, 1, [Use GNU extensions])
 
 PKG_PROG_PKG_CONFIG([0.20])
