diff -up openchange-openchange-2.3-VULCAN/libmapi/IProfAdmin.c.samba-4.13 openchange-openchange-2.3-VULCAN/libmapi/IProfAdmin.c
--- openchange-openchange-2.3-VULCAN/libmapi/IProfAdmin.c.samba-4.13	2020-11-13 13:37:00.953428673 +0100
+++ openchange-openchange-2.3-VULCAN/libmapi/IProfAdmin.c	2020-11-13 13:38:36.541453531 +0100
@@ -753,7 +753,7 @@ _PUBLIC_ enum MAPISTATUS OpenProfile(str
 _PUBLIC_ enum MAPISTATUS LoadProfile(struct mapi_context *mapi_ctx, 
 				     struct mapi_profile *profile)
 {
-	enum credentials_use_kerberos use_krb = CRED_AUTO_USE_KERBEROS;
+	enum credentials_use_kerberos use_krb = CRED_USE_KERBEROS_DESIRED;
 
 	/* Sanity checks */
 	OPENCHANGE_RETVAL_IF(!mapi_ctx, MAPI_E_NOT_INITIALIZED, NULL);
@@ -777,9 +777,9 @@ _PUBLIC_ enum MAPISTATUS LoadProfile(str
 	 * another API in the profile */
 	if (profile->kerberos) {
 		if (!strncmp(profile->kerberos, "yes", 3)) {
-			use_krb = CRED_MUST_USE_KERBEROS;
+			use_krb = CRED_USE_KERBEROS_REQUIRED;
 		} else {
-			use_krb = CRED_DONT_USE_KERBEROS;
+			use_krb = CRED_USE_KERBEROS_DISABLED;
 		}
 	}
 	/* additionally, don't set the username in the ccache if kerberos
@@ -787,13 +787,13 @@ _PUBLIC_ enum MAPISTATUS LoadProfile(str
 	 * credentials.  cli_credentials_guess probably gets the right
 	 * thing anyway in the situations where kerberos is in use */
 	if (profile->username && *(profile->username)
-	    && use_krb != CRED_MUST_USE_KERBEROS) {
+	    && use_krb != CRED_USE_KERBEROS_REQUIRED) {
 		cli_credentials_set_username(profile->credentials, profile->username, CRED_SPECIFIED);
 	}
 	if (profile->password && *(profile->password)) {
 		cli_credentials_set_password(profile->credentials, profile->password, CRED_SPECIFIED);
 	}
-	if (use_krb != CRED_AUTO_USE_KERBEROS) {
+	if (use_krb != CRED_USE_KERBEROS_DESIRED) {
 		cli_credentials_set_kerberos_state(profile->credentials, use_krb);
 	}
 
