diff -up openchange-openchange-2.3-VULCAN/configure.ac.disable-server-reqs openchange-openchange-2.3-VULCAN/configure.ac
--- openchange-openchange-2.3-VULCAN/configure.ac.disable-server-reqs	2021-06-17 21:40:49.938611315 +0200
+++ openchange-openchange-2.3-VULCAN/configure.ac	2021-06-17 21:41:18.616614841 +0200
@@ -436,7 +436,7 @@ dnl AC_SUBST(MYSQL_LIBS)
 dnl ----------------------------------------------------------------------------
 dnl Check for memcached
 dnl ----------------------------------------------------------------------------
-PKG_CHECK_MODULES([MEMCACHED], [libmemcached >= 1.0.18], [], [AC_MSG_ERROR(memcached >= 1.0.18 required to build mapistore library)])
+dnl PKG_CHECK_MODULES([MEMCACHED], [libmemcached >= 1.0.18], [], [AC_MSG_ERROR(memcached >= 1.0.18 required to build mapistore library)])
 
 dnl ----------------------------------------------------------------------------
 dnl Check for Flex
