diff --git a/src/qca_cert.cpp b/src/qca_cert.cpp
index 5e73d98..225aecd 100644
--- a/src/qca_cert.cpp
+++ b/src/qca_cert.cpp
@@ -19,6 +19,7 @@
  *
  */
 
+#include <memory>
 #include "qca_cert.h"
 
 #include "qca_publickey.h"
diff --git a/src/qca_default.cpp b/src/qca_default.cpp
index fd35702..db87e87 100644
--- a/src/qca_default.cpp
+++ b/src/qca_default.cpp
@@ -19,6 +19,7 @@
  *
  */
 
+#include <memory>
 #include "qca_core.h"
 
 #include <QMutex>
