From 09595a50f507399bf3fa8c68ed6371c42b9c4874 Mon Sep 17 00:00:00 2001
From: Julian Berman <Julian@GrayVines.com>
Date: Fri, 21 Feb 2020 09:30:05 -0500
Subject: [PATCH] Newer webcolors has a new API apparently.

---
 jsonschema/_format.py | 2 +-
 setup.cfg             | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/jsonschema/_format.py b/jsonschema/_format.py
index 281a7cfcf..f3c758c4f 100644
--- a/jsonschema/_format.py
+++ b/jsonschema/_format.py
@@ -354,7 +354,7 @@ def is_css_color_code(instance):
     def is_css21_color(instance):
         if (
             not isinstance(instance, str_types) or
-            instance.lower() in webcolors.css21_names_to_hex
+            instance.lower() in webcolors.CSS21_NAMES_TO_HEX
         ):
             return True
         return is_css_color_code(instance)
diff --git a/setup.cfg b/setup.cfg
index 250c15be1..a75c11505 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -40,11 +40,11 @@ format =
     jsonpointer>1.13
     rfc3987
     strict-rfc3339
-    webcolors
+    webcolors>=1.11.1
 format_nongpl =
     idna
     jsonpointer>1.13
-    webcolors
+    webcolors>=1.11.1
     rfc3986-validator>0.1.0
     rfc3339-validator
 
