--- src/kmcommands.cpp	2021-02-24 23:31:19.000000000 +0000
+++ ../kmcommands.cpp	2021-03-21 20:43:20.883327523 +0000
@@ -830,7 +830,7 @@
 
 KMCommand::Result KMReplyCommand::execute()
 {
-    KCursorSaver saver(Qt::WaitCursor);
+    KPIM::KCursorSaver saver(Qt::WaitCursor);
     Akonadi::Item item = retrievedMessage();
     KMime::Message::Ptr msg = MessageComposer::Util::message(item);
     if (!msg) {
@@ -890,7 +890,7 @@
     if (!msg) {
         return Failed;
     }
-    KCursorSaver saver(Qt::WaitCursor);
+    KPIM::KCursorSaver saver(Qt::WaitCursor);
 
     CreateForwardMessageJobSettings settings;
     settings.item = item;
@@ -1303,7 +1303,7 @@
 
 KMCommand::Result KMFilterActionCommand::execute()
 {
-    KCursorSaver saver(Qt::WaitCursor);
+    KPIM::KCursorSaver saver(Qt::WaitCursor);
     int msgCount = 0;
     const int msgCountToFilter = mMsgListId.count();
     ProgressItem *progressItem
@@ -1490,7 +1490,7 @@
 
 KMCommand::Result KMMoveCommand::execute()
 {
-    KCursorSaver saver(Qt::WaitCursor);
+    KPIM::KCursorSaver saver(Qt::WaitCursor);
     setEmitsCompletedItself(true);
     setDeletesItself(true);
     Akonadi::Item::List retrievedList = retrievedMsgs();
@@ -1616,7 +1616,7 @@
 
 KMCommand::Result KMTrashMsgCommand::execute()
 {
-    KCursorSaver saver(Qt::WaitCursor);
+    KPIM::KCursorSaver saver(Qt::WaitCursor);
     setEmitsCompletedItself(true);
     setDeletesItself(true);
     for (auto trashIt = mTrashFolders.begin(), end = mTrashFolders.end(); trashIt != end; ++trashIt) {
