--- inc/libcmis-c/repository.h	2019-07-07 23:51:39.613125370 +0500
+++ ../repository.h	2019-07-07 23:49:13.403133000 +0500
@@ -29,13 +29,14 @@
 #define _REPOSITORY_H_
 
 #ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <libxml/tree.h>
 
 #include "libcmis-c/libcmis-c-api.h"
 #include "libcmis-c/types.h"
+extern "C" {
+#endif
+
+
 
 LIBCMIS_C_API void libcmis_vector_repository_free( libcmis_vector_Repository_Ptr vector );
 LIBCMIS_C_API size_t libcmis_vector_repository_size( libcmis_vector_Repository_Ptr vector );
