--- kmail/plugins/common/antispamwizard.cpp	2021-02-09 12:34:22.000000000 +0000
+++ ../antispamwizard.cpp	2021-03-21 20:59:21.613385788 +0000
@@ -489,7 +489,7 @@
 void AntiSpamWizard::checkToolAvailability()
 {
     // this can take some time to find the tools
-    KCursorSaver saver(Qt::WaitCursor);
+    KPIM::KCursorSaver saver(Qt::WaitCursor);
     bool found = false;
     QVector<SpamToolConfig>::ConstIterator end(mToolList.constEnd());
     for (QVector<SpamToolConfig>::ConstIterator it = mToolList.constBegin();
